Smt/Reconstruction/Certifying/Boolean.lean

```diff
@@ -5,6 +5,25 @@ import Smt.Reconstruction.Certifying.Util
 open Lean Elab.Tactic Meta Expr Syntax
 open Nat List Classical
 
+abbrev Implies (p q : Prop) := p → q
+
+section
+
+open Lean.Macro
+
+syntax binder := "(" binderIdent term ")"
+syntax binders := binder+
+
+macro "forall " "(" xs:binders ")" b:term : term => match xs with
```

Comment:
This overrides builtin Lean notation so you may want to make it a `scoped macro` and `scoped syntax`. Then to use it, you need to explicitly `open scoped Smt.Reconstruction.Certifying.Util` (which doesn't make definitions available) or `open Smt.Reconstruction.Certifying.Util` (which opens both scoped syntax and definitions). Otherwise importing this module will change the meaning of `forall`.

Reply:
The macro is meant to be used specifically with `cvc5` proofs, and I agree that it would be risky to use it globally. I have implemented your suggestion, but I am unclear about the statement that "importing this module will alter the meaning of `forall` and `exists`." My understanding was that it is compatible with the default syntax as it requires additional parentheses around variable binders.

Reply:
[Here](https://leanprover.zulipchat.com/#narrow/stream/341532-lean4-dev/topic/Parser.20in.20another.20category.20breaks.20antiquotations) is a concrete example when something similar enough to builtin syntax breaks it. You can adopt that example to also break even when requiring extra parentheses :)

Reply:
I see... Thanks for the info.